ompi/info/info.c

```diff
@@ -116,76 +116,91 @@ int ompi_mpiinfo_init(void)
     return OMPI_SUCCESS;
 }
 
-/*
- * Fill in the MPI_INFO_ENV if using MPI3 initialization
+/**
+ * Fill in an info object with ENV info.  Used for setting
+ * MPI_INFO_ENV and by invocation of MPI_Info_create_env.
  */
-int ompi_mpiinfo_init_mpi3(void)
+
+int ompi_mpiinfo_init_env(int argc, char **argv, ompi_info_t *info)
 {
-    char *cptr, **tmp;
+    char *cptr = NULL, **tmp = NULL;
 
     /* fill the env info object */
 
     /* command for this app_context */
-    if (NULL != ompi_process_info.command) {
+    if (NULL != argv) {
+        tmp = argv;
+    } else if (NULL != ompi_process_info.command) {
         tmp = opal_argv_split(ompi_process_info.command, ' ');
-        opal_info_set(&ompi_mpi_info_env.info.super, "command", tmp[0]);
+    }
+
+    if (NULL != tmp) {
+        if (NULL != tmp[0]) {
+            opal_info_set(&info->super, "command", tmp[0]);
+        }
 
         /* space-separated list of argv for this command */
         if (1 < opal_argv_count(tmp)) {
             cptr = opal_argv_join(&tmp[1], ' ');
         } else {
-            cptr = strdup(tmp[0]);
+            if (NULL != tmp[0]) {
+                cptr = strdup(tmp[0]);
+            }
+        }
+	if (NULL == argv) {
+            opal_argv_free(tmp);
+        }
+        opal_info_set(&info->super, "argv", cptr);
```

Comment:
@hppritcha Again, I believe `cptr` can be still be NULL at this point. If that's the case, is it OK to use `cptr=NULL` with `opal_info_set()`?

Reply:
yes it should be.

Reply:
Sorry, I was not clear enough. Does it make sense to set the "argv" key if there is no relevant value to set? Perhaps it is just better to leave the info object empty? Just asking, not a complaint nor a proposal...

Reply:
I think the value for argv should be set to NULL if NULL is supplied to the call to ```MPI_Info_create_env```.   That's what I infer from the text in the standard.